File: Basic/JS/1007.js
/**
 * @author: kurumi
 * @date: 2025-09-07
 * @description: 1007 素数对猜想
 */
const rl = require("readline").createInterface({input: process.stdin});
const iter = rl[Symbol.asyncIterator]();
const readline = async () => (await iter.next()).value;

const initPrimes = (primes, n) => {
  primes[0] = false;
  primes[1] = false;
  for (let i = 2; i < n; i++) {
    for (let j = i; i * j <= n; j++) {
      primes[i * j] = false;
    }
  }
}

const countPrimes = primes => {
  let cnt = 0;
  for (let i = 2; i < primes.length; i++) {
    if (primes[i]&& primes[i - 2]) ++cnt;
  }
  return cnt;
}

void (async function() {
  const n = parseInt(await readline());
  const primes = new Array(n + 1).fill(true);
  initPrimes(primes, n)
  console.log(countPrimes(primes))
})();

File: Basic/Python/1007.py
'''
@Author: kurumi
@Date: 2025-09-07
@Description: 1007 素数对猜想
'''
from typing import List

def initPrimes(primes: List[bool], n: int):
    primes[0], primes[1] = False, False
    for i in range(2, n):
        j: int = i
        while i * j <= n:
            primes[i * j] = False
            j += 1

def countPrimes(primes: List[bool]) -> int:
    cnt: int = 0
    for i in range(2, len(primes)):
        if primes[i] == True and primes[i - 2] == True:
            cnt += 1
    return cnt

def main():
    n: int = int(input())
    primes: List[bool] = [True] * (n + 1)
    initPrimes(primes, n)
    print(countPrimes(primes))

if __name__ == "__main__":
    main()